sympy/physics/control/lti.py

```diff
@@ -1144,3 +1327,195 @@ def _eval_rewrite_as_TransferFunction(self, num, den, **kwargs):
 
     def __neg__(self):
         return Feedback(-self.num, self.den)
+
+
+class TransferFunctionMatrix(Basic):
+
+    def __new__(cls, arg, shape=None, var=None):
+        if not (isinstance(arg, (list, tuple)) and
+            (all(isinstance(i, (list, tuple)) for i in arg) or
+            all(isinstance(i, (TransferFunction, Parallel, Series)) for i in arg))):
+            raise TypeError("Unsupported type for argument of TransferFunctionMatrix.")
+
+        if shape and var:
+            if not isinstance(shape, tuple):
+                raise TypeError("Shape must be a tuple, not {}.".format(type(shape)))
+            if not isinstance(var, Symbol):
+                raise TypeError("Var must be a Symbol, not {}.".format(type(var)))
+
+        obj = super(TransferFunctionMatrix, cls).__new__(cls, arg)
+        if all(isinstance(i, (TransferFunction, Series, Parallel)) for i in arg):
+            # TFM with 1st argument of the type - [tf1, tf2, tf3, ...] or (tf1, tf2, ...)
```

Comment:
You can simplify a lot by adding
```
# make a column vector
arg = [(a,) for a in arg]
```
here. Then you can remove all the special cases.

Reply:
Oh, I see. Thanks!!

Reply:
This comment still applies

Reply:
just by adding this line of code seems to do nothing because `obj` is created above. I tried the following diff to remove the special cases, but I get a bunch of other test failures:
``` julia
                    # Now we multiply two transfer function matrices...
                    # If we have two TFMs with shape (a, b) and (b, d), respectively,
                    # then the resultant TFM will be of shape (a, d).
                    a = [[None] * arg.num_inputs for _ in range(res.num_outputs)]
                    l1, l2 = res.args[0], arg.args[0]
                    if res.num_outputs == 1:
                        l1 = [(a,) for a in l1]
                    if arg.num_inputs == 1:
                        l2 = [(a,) for a in l2]
                    for i in range(res.num_outputs):
                        for j in range(arg.num_inputs):
                            for k in range(arg.num_outputs):
                                if a[i][j] is None:     # First operation.
                                    if res.num_inputs == 1:
                                        a[i][j] = res.args[0][i] * arg.args[0][k][j]
                                    else:
                                        a[i][j] = res.args[0][i][k] * arg.args[0][k][j]
                                else:
                                    a[i][j] += res.args[0][i][k] * arg.args[0][k][j]

                    res = TransferFunctionMatrix(a, (res.num_outputs, arg.num_inputs), arg.var)
```
What do you say? I think we need to consider a bunch of other cases here also.

Reply:
@oscarbenjamin, is there any way we could escape the implementation of matrix multiplication here?

Reply:
> seems to do nothing because obj is created above.

Right, you should normalize this _before_ you construct `obj`.